src/rust/engine/fs/src/lib.rs

```diff
@@ -788,25 +808,35 @@ impl PosixFS {
     }
   }
 
-  pub fn stat_sync(&self, relative_path: PathBuf) -> Result<Stat, io::Error> {
+  pub fn stat_sync(&self, relative_path: PathBuf) -> Result<Option<Stat>, io::Error> {
```

Comment:
This seems like a surprising API change - what about asyncifying things means this is easier to couple than do separately?

Reply:
Very thorough review! Heh. I think that before I committed to async-await-ifying the calling methods, this simplified the combinators. Could undo it. But it also avoided leaking implementation details (avoiding having callers interpret errors where possible is good, imo).